### DISCUSSION TITLE
`GC.AddMemoryPressure` API usage example

### DISCUSSION THREAD

**Discussion by WeihanLi** [dotnet/runtime#93717] [Q&A] [answered]
When should we consider using the `GC.AddMemoryPressure` API? 

> Informs the runtime of a large allocation of unmanaged memory that should be taken into account when scheduling garbage collection.

https://learn.microsoft.com/en-us/dotnet/api/system.gc.addmemorypressure

From the document, it's for the unmanaged memory increase, and is there any way for the managed memory pressure? For example, could I trigger GC when there's over 100MB managed memory allocated

**Comment by Clockwork-Muse** [comment, marked as answer]
If you're writing an application and you _really_ want to kick off a GC run, just call [`GC.Collect()`](https://learn.microsoft.com/en-us/dotnet/api/system.gc.collect?view=net-7.0#system-gc-collect).  That said, it's rare you should need to manually trigger a GC run.

If you're writing a library or application that interfaces with native memory, use that API.

If you're deploying (or possibly writing) an application, and you want to make the GC possibly more aggressive, consider [setting configuration in the runtime config](https://learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector).  If you're deploying in a docker container, setting the limits on the container definition is also an option.  If you're writing a library, do neither, but you might consider informing users about memory behavior.
